### PR TITLE
Enforce that Maven/Ivy repositories use https

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds a check when a repository is added that we are using https instead of http.

### CODE REVIEW

**Comment by jasontedor** [comment]
This gives a failure like:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/jason/src/elastic/elasticsearch/qa/wildfly/build.gradle' line: 39

* What went wrong:
A problem occurred evaluating project ':qa:wildfly'.
> repository on project [wildfly] is using http for artifacts on [http://download.jboss.org]

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
```

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/1

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
Comment (mark-vieira) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: We should use `project.path` instead of `project.name` even though the full path will likely appear elsewhere in the exception message and/or stacktrace. We have a lot of ambiguous subproject names.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
Comment (mark-vieira) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: I'd use `all()` instead of `whenObjectAdded()` here. Technically, someone could register a repository _before_ applying this plugin at which point the check would not catch that. The action passed to `all()` will still apply to any future repositories, in addition to any existing ones.

**Review by mark-vieira — APPROVED** [review]
@jasontedor just a couple of minor things. Otherwise looks good.

One thing to consider though. This check will not catch instances where a repository url is later changed after initial creation. Not sure if we do that anywhere but it's still a "possibility". To be _completely_ safe we'd have to do this check immediately before dependency resolution rather than at configuration time. Not sure it's worth it, your call.

**Comment by jasontedor** [comment]
> One thing to consider though. This check will not catch instances where a repository url is later changed after initial creation. Not sure if we do that anywhere but it's still a "possibility". To be completely safe we'd have to do this check immediately before dependency resolution rather than at configuration time. Not sure it's worth it, your call.

I am okay with it, but let's discuss this in the next build sync to ensure that we have consensus around the decision. I'll merge as-is, and we can follow up if needed.

**Comment by jasontedor** [comment]
Thanks for the helpful feedback @mark-vieira!

**Pull request merged**
